src/network/local_socket_reply.cpp

```diff
@@ -139,38 +143,105 @@ void mp::LocalSocketReply::send_request(const QNetworkRequest& request, QIODevic
     auto op = request.attribute(QNetworkRequest::CustomVerbAttribute).toByteArray();
 
     // Build the HTTP method part
-    http_data += op;
-    http_data += ' ';
-    http_data += request.url().toString();
+    http_data += op + ' ' + request.url().path();
+
+    if (request.url().hasQuery())
+    {
+        http_data += "?" + request.url().query();
+    }
+
     http_data += " HTTP/1.1\r\n";
 
     // Build the HTTP Host header
-    // Host can be anything, so we'll use 'multipass'
-    http_data += "Host: multipass\r\n";
+    auto host = request.url().host().toUtf8();
```

Comment:
Hmm `toUtf8()` feels wrong here?

Reply:
I'm getting `host` as a `QByteArray` from a `QString`.  There are only 3 ways that I know of to do that: `QString::toLatin1()`, `QString::toLocal8Bit() `, and `QString::toUtf8()`.  I've tended to use `toUtf8()` when needing to do this in the past.  Is there a better way?

Reply:
Sorry, I just meant that hostname can't be UTF-8, so `::toLatin1()` feels like a better choice :)

Reply:
After discussion in chat, will change it to `toLatin1()` to look a little "weird" even though `Utf8()` works just as well.

Reply:
Done